src/Aspire.Hosting/Orchestrator/ParameterProcessor.cs

```diff
@@ -0,0 +1,202 @@
+                    IsHidden = false
+                };
+            })
+            .ConfigureAwait(false);
+        }
+    }
+
+    // Internal for testing purposes.
+    private async Task HandleUnresolvedParametersAsync()
+    {
+        await HandleUnresolvedParametersAsync(_unresolvedParameters).ConfigureAwait(false);
+    }
+
+    // Internal for testing purposes - allows passing specific parameters to test.
+    internal async Task HandleUnresolvedParametersAsync(IList<ParameterResource> unresolvedParameters)
+    {
+        // This method will continue in a loop until all unresolved parameters are resolved.
+        while (unresolvedParameters.Count > 0)
+        {
+            // First we show a notification that there are unresolved parameters.
+            var result = await interactionService.PromptMessageBarAsync(
+                 "Unresolved Parameters",
+                 "There are unresolved parameters that need to be set. Please provide values for them.",
+                 new MessageBarInteractionOptions
+                 {
+                     Intent = MessageIntent.Warning,
+                     PrimaryButtonText = "Enter Values"
+                 })
+                 .ConfigureAwait(false);
+
+            if (result.Data)
+            {
+                // Now we build up a new form base on the unresolved parameters.
+                var inputs = new List<InteractionInput>();
+
+                foreach (var parameter in unresolvedParameters)
+                {
+                    // Create an input for each unresolved parameter.
+                    inputs.Add(new InteractionInput
+                    {
+                        InputType = parameter.Secret ? InputType.SecretText : InputType.Text,
+                        Label = parameter.Name,
+                        Placeholder = "Enter value for " + parameter.Name,
+                        Required = true,
+                    });
+                }
+
+                var valuesPrompt = await interactionService.PromptInputsAsync(
+                    "Set Unresolved Parameters",
+                    "Please provide values for the unresolved parameters.",
+                    [.. inputs],
+                    new InputsDialogInteractionOptions
+                    {
+                        PrimaryButtonText = "Submit",
+                        ShowDismiss = true
+                    })
+                    .ConfigureAwait(false);
+
+                if (!valuesPrompt.Canceled)
+                {
+                    // Iterate through the unresolved parameters and set their values based on user input.
+                    for (var i = unresolvedParameters.Count - 1; i >= 0; i--)
+                    {
+                        var parameter = unresolvedParameters[i];
+                        var inputValue = valuesPrompt.Data[i].Value;
+
+                        if (inputValue is null)
```

Comment:
What about empty string? I think unentered values come back as an empty string and the value is never null.